### ISSUE TITLE
`toArray` operator like that does not wait for the stream to be complete before emitting

### ISSUE THREAD

**Issue by maxime1992** [ReactiveX/rxjs#4164]
## Feature Request

**(If this is new operator request) describe reason it should be core operator**
`reduce` waits for the observable to finish before emitting the result.
`scan` does the same job but passes values as they come.

`toArray` waits for the observable to finish before emitting the result.
I think there should be an operator similar to `toArray` that passes the array as it keeps receiving values.

To go around that issue, we can use scan an push within a new array but it feels like a core operator might be welcome.

**Comment by kwonoj** [comment]
As you already mentioned scan is verasatile operator and it should be trivial to implement behavior as you suggested. Is there specific reason you think 1. Current operator chain can't solve 2. Requires separate api surface for core?

**Comment by maxime1992** [comment]
I understand that adding a new operator might be seen as something "bad" because rxjs already has a lot of them. But thinking of `reduce` and `scan` I would've expect to have something similar for `toArray` and couldn't find it. So just raising the question.

Otherwise I guess we could also juste remove `reduce` and do `scan` + `last` for example.

**Comment by kwonoj** [comment]
For new operators, we do like to get elaborations :

1. how trivially it can be achieved with existing operators? is it requires lot of complex, or near impossible?
2. if so, can it be made as user level operator module, or it should be dealt with some of internals that should be in core operator?

What we trying to achieve / recommend is having more user level api surface anyone can opt into, instead of core library contains all operators which unnecessary burdens users for having all operators. This isn't about bytesize, more about general approachability anyone need to look into core api surface.

For that reason, we don't pair all operator set, sometimes even deprecate some while semantically it makes more sense to having it, also from v5 number of operators already being deprecated from v4.

In short, I can understand motivation but not sure given suggestion meets direction we'd like to encourage. I'd like to get some more elaborations if it can fit above 2 conditions. Otherwise, I'd suggest to publish as user level module. (https://github.com/ReactiveX/rxjs/blob/master/doc/operator-creation.md#publish-your-operator-as-a-separate-library)

**Comment by maxime1992** [comment]
Replying to both points:

1) how trivially it can be achieved with existing operators? 
```
obs$.pipe(
  scan((acc, value) => {
    acc.push(value);
    return acc;
}, [])
```

so it's not really complex IMO.

2) can it be made as user level operator module

Certainly.

It's mostly for consistency that I'm raising that issue and I totally understand your point of view :+1:

**Comment by kwonoj** [comment]
Thanks. Let me close issue for now.

**Comment by maxime1992** [comment]
I guess it's a no :sweat_smile: :man_shrugging:
